riva/clients/asr/riva_asr_client.cc

```diff
@@ -154,16 +156,27 @@ class RecognizeClient {
       }
       std::cout << std::endl;
 
-      if (word_time_offsets_) {
-        std::cout << "Timestamps: " << std::endl;
+      if (word_time_offsets_ || speaker_diarization_) {
         std::cout << std::setw(40) << std::left << "Word";
-        std::cout << std::setw(16) << std::left << "Start (ms)";
-        std::cout << std::setw(16) << std::left << "End (ms)" << std::endl;
+        if (word_time_offsets_) {
+          std::cout << std::setw(16) << std::left << "Start (ms)";
+          std::cout << std::setw(16) << std::left << "End (ms)";
+        }
+        if (speaker_diarization_) {
+          std::cout << std::setw(16) << std::left << "Speaker";
+        }
+        std::cout << std::endl;
         for (int w = 0; w < result.alternatives(0).words_size(); ++w) {
           auto& word_info = result.alternatives(0).words(w);
           std::cout << std::setw(40) << std::left << word_info.word();
-          std::cout << std::setw(16) << std::left << word_info.start_time();
-          std::cout << std::setw(16) << std::left << word_info.end_time() << std::endl;
+          if (word_time_offsets_) {
+            std::cout << std::setw(16) << std::left << word_info.start_time();
+            std::cout << std::setw(16) << std::left << word_info.end_time();
+          }
+          if (speaker_diarization_) {
+            std::cout << std::setw(16) << std::left << word_info.speaker_tag();
```

Comment:
Only write speaker tags for the first alternative.

Reply:
This loop runs only over words in first alternative.
`for (int w = 0; w < result.alternatives(0).words_size(); ++w) {`

Reply:
True.  Looks like you need to rebase.  Word timestamps and scores should be printed out for all alternatives.